sdk/python/kfp/components/executor_test.py

```diff
@@ -483,59 +615,63 @@ def test_artifact_output(self):
     }
     """
 
-        def test_func(first: str, second: str) -> Artifact:
+        def test_func(first: str, second: str, output: Output[Artifact]) -> str:
+            with open(output.path, 'w') as f:
+                f.write('artifact output')
             return first + ', ' + second
 
-        self._get_executor(test_func, executor_input).execute()
-        with open(os.path.join(self._test_dir, 'output_metadata.json'),
-                  'r') as f:
-            output_metadata = json.loads(f.read())
+        output_metadata = self.execute_and_load_output_metadata(
+            test_func, executor_input)
+
         self.assertDictEqual(
             output_metadata, {
                 'artifacts': {
-                    'Output': {
+                    'output': {
                         'artifacts': [{
                             'metadata': {},
                             'name': 'output',
                             'uri': 'gs://some-bucket/output'
                         }]
                     }
+                },
+                'parameterValues': {
+                    'Output': 'Hello, World'
```

Comment:
Another thought on this, what if user func is
```
def foo(Output: OutPath(str)) -> str:
     with Open(Output, 'w') as f:
            f.write('first')
     return 'second'
```
Curiously if we throw an error of 'key already exists' or one value overrides the other.

Reply:
For the signature

```python
@dsl.component
def my_comp(Output: dsl.OutputPath(str)) -> str:
    ...
```

we get the following IR:
```YAML
components:
  comp-test-func:
    executorLabel: exec-test-func
    outputDefinitions:
      parameters:
        Output:
          parameterType: STRING
        Output_2:
          parameterType: STRING
```

We should probably throw an exception since this behavior isn't well known/documented, making it challenging to know how to pass each output to downstream tasks. Do you agree?

Reply:
Yes, totally! 
This change is probably out of scope for the current PR.

Reply:
I will make a work item for this